sconstruct

```diff
@@ -165,26 +167,38 @@ Export('outFilePrefix')
 outputDir=Dir(env['outputDir'])
 Export('outputDir')
 
-# An action to sign an executable with certFile.
-# we encrypt with SHA256 as this is the minimum required by the Windows Store for appx packages 
-signExecCmd = ["signtool", "sign", "/fd", "SHA256", "/f", certFile]
-if certPassword:
-	signExecCmd.extend(("/p", certPassword))
-if certTimestampServer:
-	signExecCmd.extend(("/tr", certTimestampServer, "/td", "SHA256"))
-def signExec(target,source,env):
-	print([str(x) for x in target])
-	#sys.exit(1)
-	# #3795: signtool can quite commonly fail with timestamping, so allow it to try up to 3 times with a 1 second delay between each try. 
-	res=0
-	for count in range(3):
-		res=env.Execute([signExecCmd+[target[0].abspath]])
-		if not res:
-			return 0 # success
-		time.sleep(1)
-	return res # failed
-#Export via scons environment so other libraries can be signed
-env['signExec']=signExec
+assert not (apiSigningToken and certFile), "Cannot specify signing with both API token (cloud) and local certificate"
+if apiSigningToken:
+	# Code signing with SignPath HSM 
+	def signExecApi(target, source, env):
+		# Workaround to skip files that we don't want to sign
+		if len(target)>1:

```

Comment:
```suggestion
		if len(target) > 1:
```

Reply:
are we sure we don't want to sign multiple files here? this seems like a red flag if the other path doesn't perform this check

Reply:
I know and it's been doing my head in. So many signing requests get made without this check, and for unnecessary files like build\x86\IAccessible2proxy.dll.pdb. I think this work may have exposed a bug in the previous signing pipeline. Speaking to Mick, with this workaround check in place, we still end up signing all the files we want to. It's like the signing calls for individual files are correct and the other signing calls where several files are requested are done in error.

Reply:
New version can sign multiple files.